### PR TITLE
Support for storage of larger token amounts

### LINKED ISSUE #236: SequelizeValidationError: "3e+22" is not a valid bigint
**Issue**
Sqlite2 Bigint support has support till 19 digits. This is an issue if user wants to do stake and mint for large amount value.

**Proposed solution**
Sqlite3 has decimal datatype support. Digits precision is configurable.
In facilitator we are supporting till 32 digits.

```
amount: {
          type: DataTypes.DECIMAL(32),
          allowNull: false,
          validate: {
            min: 0,
          },
        },
```

### PR DESCRIPTION
**Pull request by abhayks1**
**Issue**:
Sqlite2 Bigint support has support till 19 digits. This is an issue if user wants to do stake and mint for large amount value.

**Solution**
Sqlite3 has decimal datatype support. Digits precision is configurable.
In facilitator we are supporting till 32 digits.
```
amount: {
          type: DataTypes.DECIMAL(32),
          allowNull: false,
          validate: {
            min: 0,
          },
        },
```

Fixes #236

### CODE REVIEW

File: src/repositories/GatewayRepository.ts [hunk 1]
```diff
@@ -118,10 +118,11 @@ export default class GatewayRepository extends Subject<Gateway> {
           },
         },
         bounty: {
-          type: DataTypes.BIGINT,
+          type: DataTypes.DECIMAL(32),
           allowNull: false,
           validate: {
             min: 0,
+            max: 99999999999999999999999999999999,
```
Comment (gulshanvasnani) [inline src/repositories/GatewayRepository.ts]: Make a variable for MAX_SUPPORTED_VALUE

File: test/repositories/MessageRepository/save.test.ts [hunk 2]
```diff
@@ -126,4 +127,31 @@ describe('MessageRepository::save', (): void => {
 
     Util.assertMessageAttributes(updatedMessage, message);
   });
+
+  it('should fail when gas price is higher than supported amount', async (): Promise<void> => {
```
Comment (gulshanvasnani) [inline test/repositories/MessageRepository/save.test.ts]:
```suggestion
  it('should fail when gas price is higher than supported number', async (): Promise<void> => {
```

File: src/repositories/GatewayRepository.ts (hunk 1 above)
Comment (abhayks1) [reply src/repositories/GatewayRepository.ts]: Done!

File: test/repositories/MessageRepository/save.test.ts (hunk 2 above)
Comment (abhayks1) [reply test/repositories/MessageRepository/save.test.ts]: Done!

File: src/repositories/MessageRepository.ts [hunk 3]
```diff
@@ -151,17 +152,19 @@ export class MessageRepository extends Subject<Message> {
           allowNull: true,
         },
         gasPrice: {
-          type: DataTypes.BIGINT,
+          type: DataTypes.DECIMAL(32),
           allowNull: true,
           validate: {
             min: 0,
+            max: 99999999999999999999999999999999,
```
Comment (gulshanvasnani) [inline src/repositories/MessageRepository.ts]:
```suggestion
            max: MAX_VALUE,
```

File: src/repositories/MessageTransferRequestRepository.ts [hunk 4]
```diff
@@ -112,10 +113,11 @@ export default class MessageTransferRequestRepository extends Subject<MessageTra
           unique: true,
         },
         amount: {
-          type: DataTypes.BIGINT,
+          type: DataTypes.DECIMAL(32),
           allowNull: false,
           validate: {
             min: 0,
+            max: 99999999999999999999999999999999,
```
Comment (gulshanvasnani) [inline src/repositories/MessageTransferRequestRepository.ts]:
```suggestion
            max: MAX_VALUE,
```

File: src/repositories/MessageTransferRequestRepository.ts [hunk 5]
```diff
@@ -127,17 +129,19 @@ export default class MessageTransferRequestRepository extends Subject<MessageTra
           },
         },
         gasPrice: {
-          type: DataTypes.BIGINT,
+          type: DataTypes.DECIMAL(32),
           allowNull: false,
           validate: {
             min: 0,
+            max: 99999999999999999999999999999999,
```
Comment (gulshanvasnani) [inline src/repositories/MessageTransferRequestRepository.ts]:
```suggestion
            max: MAX_VALUE,
```

File: test/repositories/MessageRepository/save.test.ts [hunk 6]
```diff
@@ -58,8 +59,8 @@ describe('MessageRepository::save', (): void => {
     gatewayAddress = '0x0000000000000000000000000000000000000001';
     sourceStatus = MessageStatus.Declared;
     targetStatus = MessageStatus.Declared;
-    gasPrice = new BigNumber(100);
-    gasLimit = new BigNumber(200);
+    gasPrice = new BigNumber('30000000000000000000000000000000');
```
Comment (gulshanvasnani) [inline test/repositories/MessageRepository/save.test.ts]: I would suggest to use **MAX_VALUE** in test-cases as well

**Comment by abhayks1** [comment]
PR is available for review..

**Pull request merged**